plugins/modules/win_dns_client.ps1

```diff
@@ -194,9 +196,18 @@ Function Get-RegistryNameServerInfo {
                     NameServerBadFormat = $false
                 }
 
-                if (($ns = Get-OptionalProperty -InputObject $iprop -Name $items.DhcpNameServer)) {
-                    $famInfo.EffectiveNameServers = $famInfo.DhcpAssignedNameServers = $ns.Split(' ')
-                }
+		# IPv6 are stored as 16 bytes REG_BINARY values. If multiple IPv6 are configured
+		# those ips are contiguous
+		$famInfo.EffectiveNameServers = $famInfo.DhcpAssignedNameServers =
+		@(if ($ns -is [System.Object[]]) {
+		for ($i = 0; $i -lt $ns.Length; $i += $items.BinaryLength) {
+		    [byte[]]$ipBytes = $ns[$i..($i + $items.BinaryLength - 1)]
+		    (New-Object -TypeName System.Net.IPAddress -ArgumentList @(, $ipBytes)).IPAddressToString
+		}
+		# IPv4 are stored as space delimited string properties in the registry
+		} else {
+		    $ns.Split(' ')
+		})
```

Comment:
you need to put the conditional back in here (alignment also seems off... please check). Basically, the part you're adding here should replace the _inside_ contents of the conditional (`if (($ns = ....`).

```suggestion
            if (($ns = Get-OptionalProperty -InputObject $iprop -Name $items.DhcpNameServer)) {
		# IPv6 are stored as 16 bytes REG_BINARY values. If multiple IPv6 are configured
		# those ips are contiguous
		$famInfo.EffectiveNameServers = $famInfo.DhcpAssignedNameServers =
		@(if ($ns -is [System.Object[]]) {
		for ($i = 0; $i -lt $ns.Length; $i += $items.BinaryLength) {
		    [byte[]]$ipBytes = $ns[$i..($i + $items.BinaryLength - 1)]
		    (New-Object -TypeName System.Net.IPAddress -ArgumentList @(, $ipBytes)).IPAddressToString
		}
		# IPv4 are stored as space delimited string properties in the registry
		} else {
		    $ns.Split(' ')
		})
            }    
```